src/sentry/static/sentry/app/views/groupDetails/shared/groupTagValues.jsx

```diff
@@ -47,46 +47,42 @@ const GroupTagValues = createReactClass({
     }
   },
 
-  fetchData() {
+  async fetchData() {
     const {params, query} = this.props;
 
     this.setState({
       loading: true,
       error: false,
     });
 
-    this.api.request(`/issues/${params.groupId}/tags/${params.tagKey}/`, {
-      query,
-      success: data => {
-        this.setState({
-          tagKey: data,
-          loading: this.state.tagValueList === null,
-        });
-      },
-      error: error => {
-        this.setState({
-          error: true,
-          loading: false,
-        });
-      },
-    });
-
-    this.api.request(`/issues/${params.groupId}/tags/${params.tagKey}/values/`, {
-      query,
-      success: (data, _, jqXHR) => {
-        this.setState({
-          tagValueList: data,
-          loading: this.state.tagKey === null,
-          pageLinks: jqXHR.getResponseHeader('Link'),
-        });
-      },
-      error: error => {
-        this.setState({
-          error: true,
-          loading: false,
-        });
-      },
-    });
+    const promises = [
```

Comment:
I refactored this because the original did not work in tests. However, now that I'm looking at it, I think we could have used the `setState` callback form to get it working properly in tests. (The problem was that it was stuck with `loading: true` because loading state was dependent on the other request data to be set).